src/components/Explore/Sidebar.tsx

```diff
@@ -18,8 +20,18 @@ import Filters from './Filters';
 
 const Sidebar = () => {
     const dataActionDispatcher = React.useContext(DataActionDispatcherContext);
-    const { selectedStation } = React.useContext(DataStateContext);
+    const { selectedStation, stationsList } = React.useContext(DataStateContext);
     const selectedStationDetails = useStationDetails(selectedStation?.name);
+    const onNavigate = (selectedStationName: string, navigate_to: string) => {
+        const index: number = stationsList.map((station) => station.name).indexOf(selectedStationName);
```

Comment:
Same suggestion here as above about replacing `map` and `indexOf` with `finxIndex`.